src/main/java/de/tum/in/www1/artemis/service/ProgrammingExerciseGradingService.java

```diff
@@ -419,10 +438,21 @@ private double calculateStaticCodeAnalysisPenalty(List<Feedback> staticCodeAnaly
      * @param successfulTestCases test cases with positive feedback.
      * @param allTests of the given programming exercise.
      */
-    private void updateResultString(Result result, Set<ProgrammingExerciseTestCase> successfulTestCases, Set<ProgrammingExerciseTestCase> allTests) {
-        // Create a new result string that reflects passed, failed & not executed test cases.
-        String newResultString = successfulTestCases.size() + " of " + allTests.size() + " passed";
-        result.setResultString(newResultString);
+    private void updateResultString(Result result, Set<ProgrammingExerciseTestCase> successfulTestCases, Set<ProgrammingExerciseTestCase> allTests, ProgrammingExercise exercise) {
+        if (result.getAssessmentType() == AssessmentType.AUTOMATIC) {
+            // Create a new result string that reflects passed, failed & not executed test cases.
+            String newResultString = successfulTestCases.size() + " of " + allTests.size() + " passed";
+            result.setResultString(newResultString);
+        }
+        else {
+            // Calculate different scores for totalScore calculation and set resultString for manual results
+            double maxScore = exercise.getMaxScore();
+            double bonusPoints = Optional.ofNullable(exercise.getBonusPoints()).orElse(0.0);
+            double calculatedScore = programmingAssessmentService.calculateTotalScore(result.getFeedbacks());
+            double totalScore = programmingAssessmentService.calculateTotalScore(calculatedScore, maxScore + bonusPoints);
```

Comment:
It seems a bit off to calculate the score again for manual assessments in this method. The method should just update the result String and do nothing else.

Reply:
The problem was that the result score calculation just included the automatic feedback, so it was falsely calculated in case of a manual result. Therefore I recalculated it like it is done for the Assessment.